lib/internal/Magento/Framework/Data/Collection/AbstractDb.php

```diff
@@ -505,6 +666,9 @@ protected function _renderOrders()
     {
         if (!$this->_isOrdersRendered) {
             foreach ($this->_orders as $field => $direction) {
+                if (in_array(strtoupper($field), self::SQL_RESERVED_WORDS)) {
```

Comment:
Hi @fredden,
Since this piece of code will be executed many-many times in Magento, it should be optimized.
To optimize the solution there, we should use `isset`.
https://stackoverflow.com/a/13483548/1839611
https://3v4l.org/Osndt

Reply:
Thanks for the suggestion. I have changed the `in_array()` call to `isset()` instead.

For testing, I could not find an existing unit test for this file. Would you suggest I create a new unit test, or modify another to cover this? (If so, which unit test would you suggest modifying?)

Reply:
@fredden pay attention that data structure should change as well, the values should be as a keys to array

Reply:
@ihor-sviziev, it looks like using `isset()` is significantly slower than `in_array()` in this case. I'm investigating other options to optimise this code.
https://3v4l.org/XUFoA#v8.2.5

Reply:
I thought to not use array_flip on each iteration, but implement the well prepared array from the beginning, so that it won't affect performance

Reply:
Here is an example: https://3v4l.org/hfjpa#v8.2.5